File: problems/problem 2/test_cases.java
public class test_cases {
    private static int failures = 0;

    private static void check(String[] arr, String expected) {
        String output = solution.longestCommonPrefix(arr);
        if (output.equals(expected)) {
            System.out.println("PASS");
        } else {
            System.out.println("FAIL Expected=" + expected + " Got=" + output);
            failures++;
        }
    }

    public static void main(String[] args) {
        check(new String[]{"flower","flow","flight"}, "fl");
        check(new String[]{"interspecies","interstellar","interstate","internet"}, "inter");
        check(new String[]{"dog","racecar","car"}, "");
        check(new String[]{"abcd","abc"}, "abc");

        if (failures > 0) System.exit(1);
        else System.exit(0);
    }
}

File: problems/problem 2/test_cases.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char* longestCommonPrefix(char **strs, int n);

int failures = 0;

void check(char *arr[], int n, char *expected) {
    char *output = longestCommonPrefix(arr, n);
    if (strcmp(output, expected) == 0) {
        printf("PASS\n");
    } else {
        printf("FAIL Expected=%s Got=%s\n", expected, output);
        failures++;
    }
}

int main() {
    char *arr1[] = {"flower","flow","flight"};
    check(arr1, 3, "fl");

    char *arr2[] = {"interspecies","interstellar","interstate","internet"};
    check(arr2, 4, "inter");

    char *arr3[] = {"dog","racecar","car"};
    check(arr3, 3, "");

    char *arr4[] = {"abcd","abc"};
    check(arr4, 2, "abc");

    if (failures > 0) return 1;
    return 0;
}
